### DISCUSSION TITLE
Control a Potentiometer with Servo

### DISCUSSION THREAD

**Discussion by marcolino7** [arendst/Tasmota#13935] [Q&A] [answered]
Hi,
I have heater on my house, the power of the Heater is controlled by a potentiometer. I just control it with and Arduino and a Servomotor, coupled to potentiometer mcanically with a custom support.
The Servo rotate of 180°, potentiometer little more, but the lower zone of the potentiometer do not intrest me.
I would like to mote to new control board with ESP-32 and Tasmota. I am doing test, but I am not able to control a servo G9.
What I need is to send to potentiometer the exact position between 0 and 255 with a command.
How can achieve this? Do you have some suggestion? Documentation is a little bit confusionary. I do not need shutter control, for me would be enought to send a command in example, PWM1 0 and then servo goes to 0 degrees, PWM1 255 and servo goes to 180 degrees.

Thanks a lot

**Comment by barbudor** [comment]
You can define a [GPIO](https://tasmota.github.io/docs/ADC/) as an "ADC Range" type
With command [AdcParams](https://tasmota.github.io/docs/Commands/#adcparams) you can map the values from the ADC (0..1023 on ESP8266, 0..4096 on ESP32) to a range of interrest
Then a [rule](https://tasmota.github.io/docs/Rules) such as `on Analog#Range1 do PWM %value% endon` will do it

**Reply by sfromis** [reply]
As I read it, the intention is to use a servo to twist a potentiometer on a heater, not to sense the setting.

**Reply by marcolino7** [reply]
> As I read it, the intention is to use a servo to twist a potentiometer on a heater, not to sense the setting.

Yes this is my intent. I need to rotate potentiometer to heater

**Comment by ascillato2** [comment, marked as answer]
How to control servos:

https://github.com/arendst/Tasmota/discussions/10443
